backend/app/controllers/api/v1/instructor/ddahs_controller.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+class Api::V1::Instructor::DdahsController < ApplicationController
+    before_action :validate_instructor
+
+    # GET /sessions/:session_id/ddahs
+    def index
+        render_success Ddah
+                           .by_session(params[:session_id])
+                           .by_instructor(@active_instructor.id)
+    end
+
+    # GET /ddahs/:ddah_id
+    def show
+        find_ddah
+        unless @ddah
+            render_error(
+                message:
+                    "Instructor '#{
+                        @active_instructor.utorid
+                    }' does not have permission to access Ddah with 'id'=#{
+                        params[:id]
+                    }"
+            )
+        end
+        render_success @ddah
+    end
+
+    # POST /ddahs
+    def create
+        find_or_create
+
+        # It is possible that `Ddah.find(...)` didn't find a DDAH, which means we passed
+        # an invalid id field and we didn't pass an `assignment_id` field. We cannot continue
+        # in this case.
+        unless @ddah
+            render_error(
+                message:
+                    'Cannot create a DDAH without an assignment_id OR improper permissions to access/create a DDAH'
+            ) && return
+        end
+
+        # Since `update` referrs to `@ddah`, and we either have an existing or a new (unsaved)
+        # `@ddah`, it is safe to just call `update` here.
+        update
+    end
+
+    private
+
+    def validate_instructor
+        @active_user = ActiveUserService.active_user request
+
+        # If we're here, we have `instructor` permissions, but that doesn't mean
+        # we actually *are* an instructor.
+        @active_instructor = Instructor.find_by(utorid: @active_user.utorid)
+        render_error(message: 'Not an instructor') unless @active_instructor
+    end
+
+    def find_or_create
+        @ddah = nil
```

Comment:
This is more of a question rather than a review.
Is it a convention to use this instance variable style to manage responses like finding an instructor/ddah? Why are we not using return values instead? I just think it is a little bit confusing since these variables are not even defined upon class initialization, which makes it a bit hard to track, so I assume it is a Ruby thing.

Reply:
Using instance variables is a form of caching, so we don't need to redo database queries.